### PR TITLE
Feature/bookmark favicons

### PR DESCRIPTION
**Pull request by CDRussell**
Task/Issue URL: https://app.asana.com/0/547792610048271/532522547697235
Tech Design URL: 
CC: 

**Description**:
Add favicon support for bookmark screen

**Steps to test this PR**:
1. Launch bookmark screen and ensure bookmarks load a favicon icon
1. Edit a bookmark (ex: change from facebook.com to twitter.com) and ensure favicon updates
1. Edit a bookmark to contain invalid url (eg: `https:/asdf`) and ensure the globe is used as a placeholder

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

**Comment by brindy** [comment]
Looks good. I notice the SERP uses a slightly different URL, `/i/` as opposed to `/ip3/`- just want to check if there's any implications of that, but otherwise it's good.

**Comment by brindy** [comment]
@nilnilnil any thoughts on using `/ip3/` vs `/I/` as the path for the favicons?

**Comment by nilnilnil** [comment]
@brindy it should be what the site uses. That looks like ip3. Here is an example src reference from the SERP: `//icons.duckduckgo.com/ip3/www.overstock.com.ico`

File: app/src/main/java/com/duckduckgo/app/global/UriExtension.kt [hunk 1]
```diff
@@ -56,3 +56,11 @@ fun Uri.toDesktopUri(): Uri {
         this
     }
 }
+
+private const val faviconBaseUrlFormat = "https://icons.duckduckgo.com/ip3/%s.ico"
```
Comment (nilnilnil) [inline app/src/main/java/com/duckduckgo/app/global/UriExtension.kt]: This is the right base, yes.

Comment (brindy) [reply app/src/main/java/com/duckduckgo/app/global/UriExtension.kt]: Thanks!

**Pull request merged**